File: Domain/Entities/Activity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace Kaizen.Domain.Entities
{
    public class Activity : RequestBase, ICloneable
    {
        public List<ActivityEmployee> ActivitiesEmployees { get; set; }
        public List<ActivityService> ActivitiesServices { get; set; }

        [NotMapped]
        public List<Employee> Employees { get; set; }
        [NotMapped]
        public List<Service> Services { get; set; }

        public object Clone()
        {
            Activity activity = new Activity
            {
                Date = Date,
                Client = Client,
                ClientId = ClientId,
                State = State,
                Periodicity = Periodicity,
                ActivitiesEmployees = new List<ActivityEmployee>(ActivitiesEmployees.Select(a => new ActivityEmployee { EmployeeId = a.EmployeeId })),
                ActivitiesServices = new List<ActivityService>(ActivitiesServices.Select(a => new ActivityService { ServiceCode = a.ServiceCode }))
            };

            activity.ActivitiesEmployees.ForEach(a => a.Activity = activity);
            activity.ActivitiesServices.ForEach(a => a.Activity = activity);

            return activity;
        }
    }
}

File: Kaizen/ClientApp/src/app/modules/activity-schedule/models/activity.ts
import { Client } from '@modules/clients/models/client';
import { Employee } from '@modules/employees/models/employee';
import { PeriodicityType } from '@modules/service-requests/models/periodicity-type';
import { Service } from '@modules/services/models/service';
import { ActivityState } from './activity-state';

export interface Activity {
	code?: number;
	date: Date;
	state: ActivityState;
	clientId: string;
	periodicity: PeriodicityType;

	employeeCodes: string[];
	serviceCodes: string[];

	employees?: Employee[];
	services?: Service[];
	client?: Client;
}
